test/dynamo/test_exceptions.py

```diff
@@ -411,6 +411,21 @@ def test_speculation_exception(self):
         with self.assertRaises(SpeculationLogDivergence):
             log.next("bad", 58, "bad", Instruction(2, "different", 2, 2))
 
+    def test_raise_GeneratorExit(self):
+        # GeneratorExit does not inherit from Exception
+        @torch.compile(backend="eager", fullgraph=True)
+        def fn(t):
+            try:
+                raise GeneratorExit
+            except Exception:
+                return t.sin()
+            except BaseException:
+                return t.cos()
+
+        t = torch.randn(2)
+        y = fn(t)
+        self.assertEqual(y, t.cos())
+
```

Comment:
Can you remind me how we ban (graph break) on generators that already have their state advanced before with hit torch.compile?

Like, how does the following work:
```py
def firstn(n):
    num = 0
    while num < n:
        yield num
        num += 1
 
@torch.compile(backend="eager", fullgraph=True)
def f(x, gen):
    val = next(gen)
    return val, x.sin()

gen = firstn(10)
next(gen)
f(torch.tensor(1.), gen)
```

Reply:
I think the answer is that we only support generator objects that are created from within torch.compile. If that is true... we should probably rename GeneratorObjectVariable to... LocalGeneratorObjectVariable or something

Reply:
> Can you remind me how we ban (graph break) on generators that already have their state advanced before with hit torch.compile?

There's a check on user_defined.py:786-789 that raises an exception if the generator is not local